akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorContextAdapter.scala

```diff
@@ -102,6 +103,30 @@ import scala.concurrent.duration._
  */
 @InternalApi private[typed] object ActorContextAdapter {
 
+  // just to get access to the class context
+  private final class TrickySecurityManager extends SecurityManager {
+    def getClassStack: Array[Class[_]] = getClassContext
+  }
+
+  private def getLoggerClass(system: ExtendedActorSystem): Class[_] = {
+    // FIXME use stack walker API when we no longer need to support Java 8
+    val trace = new TrickySecurityManager().getClassStack
+    var suitableClass: OptionVal[Class[_]] = OptionVal.None
+    var idx = 2 // skip this method and ctx.log right away
+    while (suitableClass.isEmpty && idx < trace.length) {
+      val clazz = trace(idx)
+      val name = clazz.getName
+      if (!name.startsWith("scala.runtime") && !name.startsWith("akka.actor.typed.internal"))
+        system.dynamicAccess.getClassFor[AnyRef](name) match {
+          case Success(clazz) ⇒ suitableClass = OptionVal.Some(clazz)
+          case _              ⇒ // just ignore, we'll hopefully find a class we can load further down
+        }
+      idx += 1
+    }
+
+    suitableClass.getOrElse(classOf[Behavior[_]])
```

Comment:
I think we should be defensive and catch exceptions around this, one never knows with class loaders and security managers